### PR TITLE
[1LP][RFR] Added the ability for any object to have a list destinations fn

### PR DESCRIPTION
**Pull request by psav**
* This function spans all the navigators and gives a dict of
  implementation name and navigations

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 4bbdcc219ea9e226248e9cd6f23aa0c9b5e3b352:

`cfme/utils/appliance/__init__.py`:
- :bangbang: Line 2971:10: **E121** *continuation line under-indented for hanging indent*
- :bangbang: Line 2975:1: **E303** *too many blank lines (3)*

`cfme/utils/appliance/implementations/__init__.py`:
- :heavy_exclamation_mark: Line 58:20: **W292** *no newline at end of file*

`cfme/utils/appliance/implementations/ui.py`:
- :heavy_exclamation_mark: Line 537:18: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/utils/appliance/__init__.py [hunk 1]
```diff
@@ -2963,6 +2963,14 @@ def create_view(self, view_class, o=None, override=None):
         return self.appliance.browser.create_view(
             view_class, additional_context={'object': new_obj})
 
+    def list_destinations(self):
```
Comment (jamesooden) [inline cfme/utils/appliance/__init__.py]: **REQUIRED:** Add a docstring.   In particular it took me a moment to discern what these were destinations to, so try to clarify that.

**Comment by dajoRH** [comment]
Lint report for commit 18c1be9513c54175b41b9a4885f127a5493766a9:

`cfme/utils/appliance/__init__.py`:
- :bangbang: Line 2999:10: **E121** *continuation line under-indented for hanging indent*
- :bangbang: Line 3003:1: **E303** *too many blank lines (3)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/utils/appliance/__init__.py [hunk 2]
```diff
@@ -2991,6 +2991,17 @@ def create_view(self, view_class, o=None, override=None):
...
+        """This function
```
Comment (rlbabyuk) [inline cfme/utils/appliance/__init__.py]: Required: Please update docstirng to be more accurate

**Pull request merged**